beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -1421,13 +1423,52 @@ impl<T: BeaconChainTypes> ExecutionPendingBlock<T> {
             });
         }
 
+        /*
+         * Apply the block's attestations to fork choice.
+         *
+         * We're running in parallel with the payload verification at this point, so this is
+         * free real estate.
+         */
+        let current_slot = chain.slot()?;
+        let mut fork_choice = chain.canonical_head.fork_choice_write_lock();
```

Comment:
This constitutes a "bounce" of the fork choice write lock, as we need to re-obtain it later to apply the block. I think this is acceptable as there's likely to be a large real-time delay between the two locking events due to this running during payload verification.